edge/pkg/eventbus/common/util/common_test.go

```diff
@@ -1,88 +1,224 @@
+			keys:          []string{"key1"},
+			disinfo:       map[string]interface{}{"key2": "value2"},
+			expectedError: fmt.Errorf("key not found"),
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			err := CheckKeyExist(tt.keys, tt.disinfo)
+			if !reflect.DeepEqual(err, tt.expectedError) {
+				t.Errorf("common.CheckKeyExist() error = %v, expectedError =  %v", err, tt.expectedError)
+			}
+		})
+	}
 }
 
-func (f *fakeMqttClient) AddRoute(topic string, callback MQTT.MessageHandler) {
-}
+//TestCheckClientToken checks client token received
+func TestCheckClientToken(t *testing.T) {
+	tests := []struct {
+		name          string
+		token         MQTT.Token
+		expectedError error
+	}{
+		{
+			name:          "TestCheckClientToken: Client Token with no error",
+			token:         MQTT.NewClient(clientOptions).Connect(),
+			expectedError: nil,
+		},
+		{
+			name:          "TestCheckClientToken: Client token created with error",
+			token:         MQTT.NewClient(HubclientInit("tcp://127.0.0:8000", "12345", "", "")).Connect(),
+			expectedError: fmt.Errorf("Network Error : dial tcp: lookup 127.0.0: no such host"),
+		},
+	}
 
-func (f *fakeMqttClient) OptionsReader() MQTT.ClientOptionsReader {
-	return MQTT.ClientOptionsReader{}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			fmt.Printf("something")
+			rs, err := CheckClientToken(tt.token)
+			fmt.Printf("rs  =  %v", rs)
+			rs, err = CheckClientToken(tt.token)
+			fmt.Printf("rs  =  %v", rs)
+			if !reflect.DeepEqual(err, tt.expectedError) {
+				t.Errorf("common.CheckClientToken() error = %v, expectedError =  %v", err, tt.expectedError)
+			}
+		})
+	}
 }
 
+//TestPathExist checks the functionality of PathExist function
 func TestPathExist(t *testing.T) {
-	fakePath := "/"
-	result := PathExist(fakePath)
-	common.AssertTrue(t, result, "result is not true")
-}
-
-func TestCheckKeyExist(t *testing.T) {
-	fakeKey := []string{"key1"}
-	fakeInfo := map[string]interface{}{
-		"key1": "value1",
+	tests := []struct {
+		name string
+		path string
+		want bool
+	}{
+		{
+			name: "TestPathExist: Path Exist",
+			path: "/",
+			want: true,
+		},
+		{
+			name: "TestPathExist: Path does not Exist",
+			path: "Wrong_Path",
+			want: false,
+		},
 	}
-	result := CheckKeyExist(fakeKey, fakeInfo)
-	assert.Nil(t, result)
-}
-
-func TestCheckKeyFailed(t *testing.T) {
-	fakeKey := []string{"key1"}
-	fakeInfo := map[string]interface{}{
-		"key2": "value1",
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			if got := PathExist(tt.path); !reflect.DeepEqual(got, tt.want) {
+				t.Errorf("common.TestPathExist() got = %v, want =  %v", got, tt.want)
+			}
+		})
 	}
-	result := CheckKeyExist(fakeKey, fakeInfo)
-	assert.NotNil(t, result)
 }
 
+//TestHubclientInit checks the HubclientInit method that it returns the same clientOptions object or not
 func TestHubclientInit(t *testing.T) {
```

Comment:
rename and use camel case please. Might have to make changes in source-code as well